### ISSUE TITLE
MySQL restarted while install hence Drush tasks omitted for 2 subdomains, how to repair?   

### ISSUE THREAD

**Issue by yurg** [elmsln/elmsln#2184]
The latest install script at the latest clean Ubuntu; somehow it halted at "web components have been changed, blah-blah" message at comply.domain.tld setup, so, after minutes waiting, I've restarted MySQL. and It causes Drush to skip required(?) setup steps   for 2 subdomains: lor and comply. All other subdomains seems to work fine, no other issues while install. How do I repair only those missed? Is there any Drush commands to reinstall some subdomains only?
Have tried `drush @people si cpr --y`  no luck: 
` drush @lor si cpr --y
Drush\Sql\SqlException: Unable to find a matching SQL Class. Drush   [error]
cannot find your database connection details. in
/root/.config/composer/vendor/drush/drush/commands/sql/sql.drush.inc:597
`

and 

`drush @comply si cpr --y
You are about to CREATE the 'comply_ml' database. Do you want to continue? (y/n): y
Starting Drupal installation. This takes a while. Consider using the --notify global option.   [ok]
PHP Fatal error:  Uncaught Error: Call to undefined function db_query() in /var/www/elmsln/core/dslmcode/cores/drupal-7/includes/cache.inc:359
`

**Comment by btopro** [comment]
hmm... halted or just hangs for awhile? I'm curious if you have a real domain name when invoking this? I see `people.host.tld` from other mentions, is this the real setup your trying to make and is this a valid domain?

If you try and install with an invalid domain I can imagine long hangs during creation that should eventually finish but might take awhile. I experienced this recently with an AWS instance where the domains it was being setup against didn't exist.

If you need to make a real server with fake domains, you'll need to log into the server and modify the /etc/hosts to trick it into resolving them. This could help.  For example from vagrant you get things like:
```
10.0.18.55  courses.elmsln.local
10.0.18.55  data-courses.elmsln.local
```
A front facing and data domain for each. If you changed the 10.0.18.55 to localhost in the servers /etc/hosts and made an entry for each record needed (I believe it might generate this during the installer if you look through the logs as output in the console) then it will hopefully go through on a clean install:

you can see the full list at: https://elmsln.readthedocs.io/en/latest/development/Vagrant-Step-by-Step-setup/ under `Legacy - host file management`

**Comment by yurg** [comment]
Totally halted  I afraid; actually it has been a second attempt to run the installer, first time I've lost host connection after ~10 minutes of waiting. And it is a real domain names, were propagated  a couple of weeks ago according to content of **/domains/** folder and **drush sa** output. 
For now, it seems I've messed it all up [again] with **leafy** updates, so, basically, question is the same: any chance to repair single (or several) subdomains without running the install script from scratch?
Thank you!

**Comment by btopro** [comment]
Ok, I can reproduce this. interesting one for sure.. It has to do with our performance optimizations that we add in AFTER the installations go through. So... to undo this temporarily so that the install can run through do this.
edit `/var/www/elmsln/config/stacks/people/sites/people/ml/settings.php` and comment out the last line in the file
Then do the install
so for people
`drush @people si cpr --y`
Then put that line back in in the file. This should work for inbox, people and anything else that didn't build correctly during the installer.

**Comment by btopro** [comment]
you'll probably also need to do `drush @people en elmsln_bakery --y` afterwards for each one

**Comment by yurg** [comment]
Thank you for help, I guess reproducing haven't been easy. Have got another one  after commenting out 
**require_once DRUPAL_ROOT . '/../../shared/drupal-7.x/settings/shared_settings.php';**: 


` drush si cpr --y
You are about to CREATE the 'online_ml' database. Do you want to continue? (y/n): y
Starting Drupal installation. This takes a while. Consider using the --notify global option.                    [ok]
Exception: <ul><li>To start over, you must empty your existing database.</li><li>To install to a different      [error]
database, edit the appropriate <em>settings.php</em> file in the <em>sites</em> folder.</li><li>To upgrade an
existing installation, proceed to the <a href="http://default/update.php">update script</a>.</li><li>View your
<a href="http://default">existing site</a>.</li></ul> in
/var/www/elmsln/core/dslmcode/cores/drupal-7/includes/install.core.inc:301
`

I've cleaned the database, it hasn't help. I've deleted it: all the same. Recreated the db: the same.
Have tried to run it both from ~ and /var/www/elmsln/config/stacks/online/

**Comment by btopro** [comment]
it has to be in the installs in question. The targets for the drush commands are critical to hitting the right drupal install. There are 12 or so drupal installs out of the box cause this isn't a traditional drupal deployment (at all). `drush @people si cpr --y`

Otherwise you just hit a default and probably the online one if it asked about creating online_ml which would cause some real issues / make you end up having to run this again against `@online`

**Comment by yurg** [comment]
Sorry, not sure I completely follow, will you please tell again, from where `drush @people si cpr --y` must be executed?

**Comment by btopro** [comment]
anywhere commandline. Your previous command you copy and pasted didn't have the `@people` part in it

**Comment by yurg** [comment]
Sorry, it doesn't work: I've omitted `@people` in one of the prev. examples because tried to run it from  a folder where drupal files are located, assuming default Drush behaviour. 
Have just checked: 
`
 drush @people  si cpr --y
You are about to CREATE the 'people_ml' database. Do you want to continue? (y/n)                                                         : y
Starting Drupal installation. This takes a while. Consider using the [ok]
--notify global option.
PHP Fatal error:  Uncaught Error: Call to undefined function db_query() in /var/                                                         www/elmsln/core/dslmcode/cores/drupal-7/includes/cache.inc:359
Stack trace:
...
`
May be Drush somehow doesn't have an access to DB?

**Comment by btopro** [comment]
Sounds like that line isn't removed from the appropriate settings file. Running out of reasons not to just rebuild entirely. We install this constantly and don't run into what you are reporting. D any of the other sites respond / work?

**Comment by yurg** [comment]
Thank you for your efforts! Have just tried another one, double-checked if the line was removed from **settings.php**  , have got same results as reported earlier:
`
 drush @online  si cpr --y
You are about to CREATE the 'online_ml' database. Do you want to continue? (y/n): y
Starting Drupal installation. This takes a while. Consider using the --notify global option.                                [ok]
Exception: <ul><li>To start over, you must empty your existing database.</li><li>To install to a different database, edit   [error]
the appropriate <em>settings.php</em> file in the <em>sites</em> folder.</li><li>To upgrade an existing installation,
proceed to the <a href="http://online.mlc.org.au/update.php">update script</a>.</li><li>View your <a
href="http://online.mlc.org.au">existing site</a>.</li></ul> in
/var/www/elmsln/core/dslmcode/cores/drupal-7/includes/install.core.inc:301

`

**Comment by btopro** [comment]
but are you doing this against online? you said previously you were doing it against people. The `@` targets relate to the domain they are for

**Comment by yurg** [comment]
Sorry to cause the mess; long story short: both @people and @online are down, so I've tried against both. 
Let me, please draw the whole picture so we'll be on a same page. So:
1.  I've tried to set it up and failed, created #2137 
2. Following your advice from #2137, I've made another setup attempt, which caused #2175 to be created.
3. While #2175 was closed with unclean resolution, so #2184 was born.
4. We are currently at  #2184, where I'm trying to restore several (mean, @people, @online and others) subdomains, following closely any advice you give.  I've tried to remove the last row from settings.php at @people, no luck. Have made the same at @online: same result; both are reported back to #2184. 
The bottom line (the most recent error we can't overcome):
`drush @online si cpr --y You are about to CREATE the 'online_ml' database. Do you want to continue? (y/n): y Starting Drupal installation. This takes a while. Consider using the --notify global option. [ok] Exception: <ul><li>To start over, you must empty your existing database.</li><li>To install to a different database, edit [error] the appropriate <em>settings.php</em> file in the <em>sites</em> folder.</li><li>To upgrade an existing installation, proceed to the <a href="http://online.mlc.org.au/update.php">update script</a>.</li><li>View your <a href="http://online.mlc.org.au">existing site</a>.</li></ul> in /var/www/elmsln/core/dslmcode/cores/drupal-7/includes/install.core.inc:301
`
Regarding this the latest one: I've tried to clean up the database, as well as drop and recreate it: all 3 times it has been the same. 

Thank you for all your help, I feel we're close to nail it, let us think what might cause Drush to fail while accessing   online_ml database. Sorry for the confusion, I will work at **online** all the time, if switching before two non-working instances is so confusing. Please let me know, if it makes sense. Thank you!

**Comment by btopro** [comment]
are you running mariadb or mysql? U16 from scratch assumes mariadb, maybe it's actually got mysql going. I'm really not sure what you tell you, we don't have issues like this. If this is being installed on an existing server I would highly recommend against that as we always put elmsln on it's own server because it effectively takes over the machine.

**Comment by yurg** [comment]
According to advice you gave me at #2137, I've set up a fresh clear Ubuntu 16.4 box and ran elmsln installer right after that, no other modification to the OS have been made, so, server runs whatever elmsln installer configured it to run and I believe elmsln has a full control over the server from the very beginning. 
Do you think I should start over fresh again?

**Comment by btopro** [comment]
`PDOException: SQLSTATE[HY000] [2002] Connection refused in lock_may_be_available() (line 167 of /var/www/elmsln/core/dslmcode/cores/drupal-7/includes/lock.inc).` is not normal. 

Have you tried restarting mysql / can you verify that mysql is running in any capacity?

**Comment by yurg** [comment]
Yes, MySQL has been running at the time I've tried to execute **drush @online si cpr --y**, this is the fresh one:
`
 ps aufx | grep mysql
root      9665  0.0  0.0  11284  1012 pts/0    S+   01:41   0:00  |       \_ gre                                                      p --color=auto mysql
mysql    18607  0.1 10.8 1144232 440700 ?      Sl   Aug25   5:35 /usr/sbin/mysq                                                       d --basedir=/usr --datadir=/var/lib/mysql --plugin-dir=/usr/lib/mysql/plugin --u                                                      ser=mysql --skip-grant-tables --skip-networking --skip-log-error --open-files-li                                                      mit=65535 --pid-file=/var/run/mysqld/mysqld.pid --socket=/var/run/mysqld/mysqld.                                                      sock --port=3306
root     18608  0.0  0.0  23180  1364 ?        S    Aug25   0:00 logger -t mysq                                                       d -p daemon error
~# drush @online si cpr --y
You are about to CREATE the 'online_ml' database. Do you want to continue? (y/n): y
Starting Drupal installation. This takes a while. Consider using the --notify global option.                               [ok]
Exception: <ul><li>To start over, you must empty your existing database.</li><li>To install to a different database, edit  [error]
the appropriate <em>settings.php</em> file in the <em>sites</em> folder.</li><li>To upgrade an existing installation,
proceed to the <a href="http://online.mlc.org.au/update.php">update script</a>.</li><li>View your <a
href="http://online.mlc.org.au">existing site</a>.</li></ul> in
/var/www/elmsln/core/dslmcode/cores/drupal-7/includes/install.core.inc:301
`

**Comment by btopro** [comment]
your not using the targets correctly. Drush Site-install (short si) is for installing install profiles. each domain has it's own specific profile

`drush @online si cis --y`

`drush @people si cpr --y`

`drush @media si elmsmedia --y`

`drush @interact si icor --y`

`drush @comply si ecd --y`

**Comment by yurg** [comment]
**1. Check if settings.php was altered:** 

` tail -5 /var/www/elmsln/config/stacks/online/sites/online/ml/settings.php
$conf['cache_prefix'] = 'online_ml';
//require_once DRUPAL_ROOT . '/../../shared/drupal-7.x/settings/shared_settings.php';
`

**2. Immediately after that run drush with double-checked correct install profile:** 

` drush @online si cis --y
You are about to CREATE the 'online_ml' database. Do you want to continue? (y/n): y
Starting Drupal installation. This takes a while. Consider using the --notify global option.                               [ok]
Exception: <ul><li>To start over, you must empty your existing database.</li><li>To install to a different database, edit  [error]
the appropriate <em>settings.php</em> file in the <em>sites</em> folder.</li><li>To upgrade an existing installation,
proceed to the <a href="http://online.mlc.org.au/update.php">update script</a>.</li><li>View your <a
href="http://online.mlc.org.au">existing site</a>.</li></ul> in
/var/www/elmsln/core/dslmcode/cores/drupal-7/includes/install.core.inc:301
`

**Comment by yurg** [comment]
Run drush against @people, with settings.php left intact (mean last row is not commented)

`
 drush @people si cpr --y
You are about to CREATE the 'people_ml' database. Do you want to continue? (y/n): y
Starting Drupal installation. This takes a while. Consider using the --notify global option.                               [ok]
PHP Fatal error:  Uncaught Error: Call to undefined function db_query() in /var/www/elmsln/core/dslmcode/cores/drupal-7/includes/cache.inc:359
`

**Comment by btopro** [comment]
last row must be commented

**Comment by yurg** [comment]
Yes, it is 

1. Check if settings.php was altered:

tail -5 /var/www/elmsln/config/stacks/online/sites/online/ml/settings.php $conf['cache_prefix'] = 'online_ml'; //require_once DRUPAL_ROOT . '/../../shared/drupal-7.x/settings/shared_settings.php';

2. Immediately after that run drush with double-checked correct install profile:

drush @online si cis --y You are about to CREATE the 'online_ml' database. Do you want to continue? (y/n): y Starting Drupal installation. This takes a while. Consider using the --notify global option. [ok] Exception: <ul><li>To start over, you must empty your existing database.</li><li>To install to a different database, edit [error] the appropriate <em>settings.php</em> file in the <em>sites</em> folder.</li><li>To upgrade an existing installation, proceed to the <a href="http://online.mlc.org.au/update.php">update script</a>.</li><li>View your <a href="http://online.mlc.org.au">existing site</a>.</li></ul> in /var/www/elmsln/core/dslmcode/cores/drupal-7/includes/install.core.inc:301

**Comment by btopro** [comment]
`2. Immediately after that run drush with double-checked correct install profile:` I'm sorry I've never seen this error in 7 years of working on Drupal 7. I'm sure it's possible but I don't really know what to tell you.

**Comment by yurg** [comment]
Thank you for all your help. So, this is the conclusion: it seems it is quite tricky, if not to say impossible,  to install elmsln, even with all the help of a core developers team.  
As a side note, if I may: this idea of hands-free installer is brilliant, yet, any chance there is a "manual" installation steps,  we could use it.

**Comment by btopro** [comment]
we have dozens of installs so I think saying this is impossible is a mischaracterization. You can spin this up via vagrant locally and see this is the case. Maybe try a different host or see if when you install fresh if there are any other issues during installation. It sounds like an issue happened where ur connection was lost during install. I can't guarantee success with installing if the connection is interrupted.

We've installed this successfully on AWS - https://drupal.psu.edu/blog/post/elmsln-aws-ec2
and Digital Ocean and on limited resource systems like mini-PCs and micro-PC sticks.

https://elmsln.readthedocs.io/en/latest/development/Vagrant-Step-by-Step-setup/ is how to get up and going with Vagrant if you're just looking to explore and see that it does work.

**Comment by btopro** [comment]
maybe try a new install but using 0.9.x, the branch we're close to marking a full release on for 0.9.0
you can do this during the 1-line installer putting in 0.9.x instead of master

```
apt-get update && apt-get -y install wget git && git clone https://github.com/elmsln/elmsln.git /var/www/elmsln && cd /var/www/elmsln && git checkout 0.9.x && bash /var/www/elmsln/scripts/install/handsfree/ubuntu16/ubuntu16-install.sh elmsln ln elmsln.dev http admin@elmsln.dev yes
```

**Comment by yurg** [comment]
ok, will happily serve as a beta-tester; here's a first one: 

`
running: make
/bin/bash /tmp/pear/temp/pear-build-rootwN25z3/uploadprogress-1.0.3.1/libtool --mode=compile cc  -I. -I/tmp/pear/temp/uploadprogress -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-rootwN25z3/uploadprogress-1.0.3.1/include -I/tmp/pear/temp/pear-build-rootwN25z3/uploadprogress-1.0.3.1/main -I/tmp/pear/temp/uploadprogress -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /tmp/pear/temp/uploadprogress/uploadprogress.c -o uploadprogress.lo
libtool: compile:  cc -I. -I/tmp/pear/temp/uploadprogress -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-rootwN25z3/uploadprogress-1.0.3.1/include -I/tmp/pear/temp/pear-build-rootwN25z3/uploadprogress-1.0.3.1/main -I/tmp/pear/temp/uploadprogress -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /tmp/pear/temp/uploadprogress/uploadprogress.c  -fPIC -DPIC -o .libs/uploadprogress.o
/tmp/pear/temp/uploadprogress/uploadprogress.c: In function 'uploadprogress_php_rfc1867_file':
/tmp/pear/temp/uploadprogress/uploadprogress.c:160:31: error: 'ENFORCE_SAFE_MODE' undeclared (first use in this function)
                 int options = ENFORCE_SAFE_MODE;
                               ^
/tmp/pear/temp/uploadprogress/uploadprogress.c:160:31: note: each undeclared identifier is reported only once for each function it appears in
/tmp/pear/temp/uploadprogress/uploadprogress.c:216:28: warning: format '%d' expects argument of type 'int', but argument 6 has type 'time_t {aka long int}' [-Wformat=]
                 fprintf(F, "upload_id=%s\nfieldname=%s\nfilename=%s\ntime_start=%d\ntime_last=%d\nspeed_average=%d\nspeed_last=%d\nbytes_uploaded=%d\nbytes_total=%d\nfiles_uploaded=%d\nest_sec=%d\n",
                            ^
/tmp/pear/temp/uploadprogress/uploadprogress.c:216:28: warning: format '%d' expects argument of type 'int', but argument 7 has type 'time_t {aka long int}' [-Wformat=]
/tmp/pear/temp/uploadprogress/uploadprogress.c:216:28: warning: format '%d' expects argument of type 'int', but argument 10 has type 'long unsigned int' [-Wformat=]
/tmp/pear/temp/uploadprogress/uploadprogress.c:216:28: warning: format '%d' expects argument of type 'int', but argument 11 has type 'long unsigned int' [-Wformat=]
/tmp/pear/temp/uploadprogress/uploadprogress.c: In function 'uploadprogress_file_php_get_info':
/tmp/pear/temp/uploadprogress/uploadprogress.c:420:57: error: macro "add_assoc_string" passed 4 arguments, but takes just 3
                 add_assoc_string( return_value, k, v, 1 );
                                                         ^
/tmp/pear/temp/uploadprogress/uploadprogress.c:420:17: error: 'add_assoc_string' undeclared (first use in this function)
                 add_assoc_string( return_value, k, v, 1 );
                 ^
/tmp/pear/temp/uploadprogress/uploadprogress.c: In function 'uploadprogress_file_php_get_contents':
/tmp/pear/temp/uploadprogress/uploadprogress.c:437:19: error: 'ENFORCE_SAFE_MODE' undeclared (first use in this function)
     int options = ENFORCE_SAFE_MODE;
                   ^
/tmp/pear/temp/uploadprogress/uploadprogress.c:458:71: error: macro "php_stream_copy_to_mem" passed 4 arguments, but takes just 3
         if ((len = php_stream_copy_to_mem(stream, &contents, maxlen, 0)) > 0) {
                                                                       ^
/tmp/pear/temp/uploadprogress/uploadprogress.c:458:20: error: 'php_stream_copy_to_mem' undeclared (first use in this function)
         if ((len = php_stream_copy_to_mem(stream, &contents, maxlen, 0)) > 0) {
                    ^
/tmp/pear/temp/uploadprogress/uploadprogress.c:466:44: error: macro "RETVAL_STRINGL" passed 3 arguments, but takes just 2
             RETVAL_STRINGL(contents, len, 0);
                                            ^
/tmp/pear/temp/uploadprogress/uploadprogress.c:466:13: error: 'RETVAL_STRINGL' undeclared (first use in this function)
             RETVAL_STRINGL(contents, len, 0);
             ^
Makefile:194: recipe for target 'uploadprogress.lo' failed
make: *** [uploadprogress.lo] Error 1
ERROR: 'make' failed  mkdir: cannot create directory '/var/log/php-fpm/': File exists
setsebool:  SELinux is disabled. 
`


and install script hangs here for a couple of minutes. Soon after another one pops up

`Job for mariadb.service failed because the control process exited with error code. See "systemctl status mariadb.service" and "journalctl -xe" for details.`

there are some more escaped from screen, here's one

`Failed to restart php5-fpm.service: Unit php5-fpm.service not found.
You really should use https and invest in certs... seriously do it!
ln: failed to create symbolic link '/etc/apache2/sites-enabled/000-default.conf': File exists
ln: failed to create symbolic link '/usr/local/bin/drush-create-site/drush-create-site': File exists
ln: failed to create symbolic link '/usr/local/bin/drush-command-job/drush-command-job': File exists `

and another 

`
[Composer\Downloader\FilesystemException]                                                
Filesystem exception:                                                                    
Composer update failed: the "/root/.composer/cache" directory used to download the temp file could not be written
self-update [-r|--rollback] [--clean-backups] [--no-progress] [--update-keys] [--stable] [--preview] [--snapshot] [--set-channel-only] [--] [<version>]
 [ErrorException]
chdir(): Permission denied (errno 13) 
`

**Comment by yurg** [comment]
It's strange installer tries to restart php5-fpm while php7-fpm has been installed from the very beginning.

**Comment by yurg** [comment]
It hangs at the very same place as before: 

`
drush installing authority tool: comply
file_private_path was set to                                                    [success]
"/var/www/elmsln/config/private_files/comply/comply".
file_temporary_path was set to "/var/www/elmsln/config/tmp".                    [success]
The following extensions will be enabled: cis_service_restws
Do you really want to continue? (y/n): y
Web components have changed! Make sure that your attributes are correctly       [warning]
reflected in your site. properties changed on lrndesign-drawer from
a:5:{s:9:&quot;innerHTML&quot;;s:6:&quot;String&quot;;s:6:&quot;opened&quot;;s:6:&quot;String&quot;;s:5:&quot;align&quot;;s:6:&quot;String&quot;;s:4:&quot;icon&quot;;s:6:&quot;String&quot;;s:3:&quot;alt&quot;;s:6:&quot;String&quot;;}
to
s:133:&quot;a:5:{s:9:&quot;innerHTML&quot;;s:6:&quot;String&quot;;s:6:&quot;opened&quot;;s:6:&quot;String&quot;;s:5:&quot;align&quot;;s:6:&quot;String&quot;;s:4:&quot;icon&quot;;s:6:&quot;String&quot;;s:3:&quot;alt&quot;;s:6:&quot;String&quot;;}&quot;;
` 

Guess if I restart MySQL now, we'll return to the beginning of this conversation.

**Comment by btopro** [comment]
Are you running this as the server's root account? install HAS to be done as root

**Comment by yurg** [comment]
Yes, I'm running this as root all the time from the very beginning.

**Comment by yurg** [comment]
It hangs for a 5 minutes now, here's the latest record from  syslog: 

`
CRON[591]: (ulmusdrush) CMD (/usr/local/bin/drush-command-job/drush-command-job >> /var/www/elmsln/config/logs/drush-command-job.log)
`

**Comment by yurg** [comment]
Yet another idea: why it tries to create mysql binary logs?  

` ll /var/log/mysql
total 27540
drwxr-s---  2 mysql adm        4096 Aug 29 06:07 ./
drwxrwxr-x 12 root  syslog     4096 Aug 29 06:25 ../
-rw-rw----  1 mysql adm    28184922 Aug 29 06:15 mariadb-bin.000001
-rw-rw----  1 mysql adm          34 Aug 29 06:06 mariadb-bin.index
`

**Comment by yurg** [comment]
.. and aria_logs:

`
 ll /var/lib/mysql
total 176212
drwxr-xr-x 11 mysql mysql     4096 Aug 29 06:14 ./
drwxr-xr-x 48 root  root      4096 Aug 29 06:06 ../
-rw-rw----  1 mysql mysql    16384 Aug 29 06:06 aria_log.00000001
-rw-rw----  1 mysql mysql       52 Aug 29 06:06 aria_log_control
`

**Comment by btopro** [comment]
yeah that's a job for invoking new sites to build so thats supposed to be there in the background.

Your sure that this host doesn't already install things on the server cause it keeps saying that things already exist in your logs your copying.

Also that uploadprogress block is weird, also mariadb.service failing to start isn't a good sign obviously. Does this host have anything alternate repos that it uses for Ubuntu compiling?

**Comment by yurg** [comment]
Let me try again: it is a fresh Ubuntu box with nothing installed prior to install script run.

To be absolutely sure, I've executed [each time before install script] this command: 
`apt-get purge apache* mysql* mariad* php* && apt-get autoremove &&  apt-get autoclean &&  apt-get dist-upgrade`

and after that:

find / -name "mysql*" (php, apache2) and remove all findings, when server reboot.

**Comment by yurg** [comment]
`
 cat /etc/apt/sources.list
deb http://mirror.binarylane.com.au/ubuntu/ xenial main restricted universe multiverse
deb http://mirror.binarylane.com.au/ubuntu/ xenial-updates main restricted universe multiverse
deb [arch=amd64,i386,ppc64el] http://ftp.utexas.edu/mariadb/repo/10.1/ubuntu xenial main
# deb-src [arch=amd64,i386,ppc64el] http://ftp.utexas.edu/mariadb/repo/10.1/ubuntu xenial main
# deb-src [arch=i386,amd64,ppc64el] http://ftp.utexas.edu/mariadb/repo/10.1/ubuntu xenial main
deb http://security.ubuntu.com/ubuntu/ xenial-security main restricted universe multiverse
# deb-src [arch=amd64,ppc64el,i386] http://ftp.utexas.edu/mariadb/repo/10.1/ubuntu xenial main
# deb-src [arch=ppc64el,i386,amd64] http://ftp.utexas.edu/mariadb/repo/10.1/ubuntu xenial main
# deb-src [arch=amd64,ppc64el,i386] http://ftp.utexas.edu/mariadb/repo/10.1/ubuntu xenial main
# deb-src [arch=ppc64el,i386,amd64] http://ftp.utexas.edu/mariadb/repo/10.1/ubuntu xenial main
# deb-src [arch=amd64,i386,ppc64el] http://ftp.utexas.edu/mariadb/repo/10.1/ubuntu xenial main
# deb-src [arch=i386,ppc64el,amd64] http://ftp.utexas.edu/mariadb/repo/10.1/ubuntu xenial main
# deb-src [arch=i386,amd64,ppc64el] http://ftp.utexas.edu/mariadb/repo/10.1/ubuntu xenial main

`

**Comment by yurg** [comment]
`
 ll  /etc/apt/sources.list.d
total 16
drwxr-xr-x 2 root root 4096 Aug 17 23:21 ./
drwxr-xr-x 6 root root 4096 Aug 17 20:27 ../
-rw-r--r-- 1 root root  134 Aug 29 06:06 certbot-ubuntu-certbot-xenial.list
-rw-r--r-- 1 root root  134 Aug 29 06:06 certbot-ubuntu-certbot-xenial.list.save

`

**Comment by yurg** [comment]
What if I give you server root credentials? Believe you will use it for good  :-)

**Comment by btopro** [comment]
i mean I'd copy and paste that command in. I'm really not sure what to say, we honestly have dozens of successful deploys from master as well as 0.9.x. All i'd do is run the same command and hope my connection doesnt hang out with australia. Those AU mirrors maybe are different from US based? I'm grasping at straws here

**Comment by yurg** [comment]
I see :-)  Sorry, it's cruel to bother you any longer. Thank you anyway, it was interesting journey.  Think this  issue can be closed now.

**Comment by btopro** [comment]
maybe try a different OS or hosting provider or at least see if you can get the vagrant image running on your local machine. If you can't get the local vagrant image running then it's definitely a weird zone issue because that downloads a KNOWN working configuration and one that we develop on daily in multiple different host operating systems

**Comment by yurg** [comment]
I've made the last attempt to run it using bash -x to see details; it hangs at mariadb service restart with this error:
**2017-08-29  7:52:46 140710852757760 [ERROR] InnoDB: Unable to lock ./ibdata1, error: 11**
I'm quite certain it is something about my.cnf configuration installer creates/uses.
Any chance we can alter it before running the install script? 
I'd disabled binary logging, replacing it for "normal" logging, changed path to ibdata, if set and possible made some other adjustments preventing mysql from operate.

**Comment by yurg** [comment]
I'd also excluded "uploadprogress.so" php library, since it gives a reproducible error each time install script runs.

**Comment by btopro** [comment]
ok, getting somewhere... I wonder if it's selecting the wrong auto install for some reason... Instead of my.txt maybe try https://github.com/elmsln/elmsln/blob/master/scripts/server/my_ubunut16.txt and yes you can mod these things before they are applied

**Comment by yurg** [comment]
So, you think  it may work if I comment out this bits of **https://github.com/elmsln/elmsln/blob/master/scripts/install/root/elmsln-preinstall.sh**

`
#Add on our last bit of conf for our new Ubuntu Stuff.
if [[ $os == '2' && $dist == *"DISTRIB_RELEASE=16"* ]]; then
  cat /var/www/elmsln/scripts/server/my_ubunut16.txt >> $mycnf
fi
`

**Comment by btopro** [comment]
is it applying that version or the other? Maybe empty both of the my.txt and my_ubuntu16.txt files so that neither applies their enhancements.

We do performance optimization OOTB too so that's what's going on here as for having the files at all.

**Comment by yurg** [comment]
I've decided to try Centos installer, which seems to get less optimisation attempts and it works for now. As a side note: not really sure if it is a good idea to expose admin login and password to web: it is visible in browser for everyone looking at installation progress.

**Comment by btopro** [comment]
So it installed then that's great!

You mean the raw IP address during installation as far a what's going on in the logs under the hood? After logging into the site the INSTALL_LOG.txt is overwritten with directions on how to regain access via command line if you lost the password so it's insecure for the few seconds where you are going to go login after installing the platform.
